main.go

```diff
@@ -32,6 +32,8 @@ import (
 
 	// required for updating and validating the CRD clientset
 	_ "k8s.io/code-generator/cmd/client-gen/generators"
+	// main.go:36:2: import "sigs.k8s.io/controller-tools/cmd/controller-gen" is a program, not an importable package
```

Comment:
More info on how to avoid this: https://github.com/go-modules-by-example/index/blob/master/010_tools/README.md#tools-as-dependencies

Reply:
I was just wondering what the advantage would be of compiling this into the code vs doing `go get` inside the update-crd.sh script?

Reply:
Thanks to the `// build` tag, this dependency will not be built into your binaries 😄  this is just used to have go modules track the dependency for you as part of your module.

`go get` will also work, although you'll need to be careful to `export GO111MODULE=off` before to avoid it modifying your actual project's `go.mod` file.

Reply:
I've tried this and cannot seem to get `go test ./...` to be happy, even with `RUN go test -v $(go list ./... | grep -v /tools/)` 🤔